contracts/auth/SiweAuth: Change _domain visibility to internal (#529)
oasisprotocol/sapphire-paratime: docs/develop/authentication.md

}
```

#### A few words about the SIWE domain parameter {#siwe-domain}

> During contract deployment you have to provide **the domain** where the web
> content of your dApp will be hosted at. MetaMask will check whether the domain
> shown in the user's browser window matches the one provided in the SIWE message
> used for logging in and then warn the user, if there is a discrepancy. On the
> other side—if the SIWE message is forged to match the (exploited) web site
> domain, the on-chain [SiweAuth] message validation code will fail and prevent
> a user from obtaining a valid token.

Reviewer: ```suggestion
During contract deployment you have to provide **the domain** where the web
content of your dApp will be hosted at. MetaMask will check whether the domain
shown in the user's browser window matches the one provided in the SIWE message
used for logging in and then warn the user if there is a discrepancy. On the
other hand‒if the SIWE message is forged to match the (exploited) web site
domain, the on-chain [SiweAuth] message will fail validation and prevent the
user from obtaining a valid token.
```

When deploying your contract, the provided domain **should include the host
chunk of the [URI], including any subdomains, and optionally the port**. No
scheme (e.g. `http://`, `https://`) or path (e.g. `/my-app/login`) should be
included. If you wish to enforce authentication only on a specific
port, provide it alongside the domain, e.g. `mydomain.com:12345`. Otherwise,
MetaMask will consider any port valid.

> The visibility of `_domain` in [SiweAuth] is **`internal`**. By default, a
> public getter is implemented, so a web app can automatically obtain a domain
> name when generating the SIWE message. No setters are provided should you wish
> to keep the domain immutable. Otherwise, feel free to implement a setter in
> your contract with appropriate authentication mechanisms (e.g. `onlyOwner`
> modifier). For traceability, we also suggest to **emit an event** in case the
> domain is changed since transactions may be encrypted.

Reviewer: ```suggestion
The visibility of `_domain` in [SiweAuth] is **`internal`**. By default, a
public getter is implemented, so a web app can automatically obtain a domain
name when generating the SIWE message. No setters are provided in keeping
the domain immutable. If needed, feel free to implement a setter in
your contract with appropriate authentication mechanisms (e.g. `onlyOwner`
modifier). For traceability, we also suggest to **emit an event** when the
domain is changed as transactions may be encrypted.
```

:::info Example: Starter project

To see a running example of the TypeScript SIWE code including the Hardhat
tests, Node.js and the browser, check out the official Oasis [demo-starter]

> project. The SIWE authentication is implemented in the backend as a
> [Hardhat task], inside [unit tests] and in the frontend inside the
> [Web3 Auth provider] code.

Reviewer: ```suggestion
project. The SIWE authentication is implemented on the backend as a
[Hardhat task], in [unit tests], and on the frontend within the
[Web3 Auth provider] code.
```

:::

:::tip Sapphire TypeScript wrapper?

> While the [Sapphire TypeScript wrapper][sp-npm] offers a convenient end-to-end

Reviewer: ```suggestion
While the [Sapphire TypeScript wrapper][sp-npm] offers convenient end-to-end
```

> encryption for contract calls and transactions, it is not mandatory for SIWE
> security, if you trust your Web3 endpoint. The token generation is performed
> inside the Sapphire's TEE and the communication with your Web3 endpoint is
> secured via HTTPS.

Reviewer: ```suggestion
encryption for contract calls and transactions, using the TypeScript wrapper
is not required for SIWE security if you trust your Web3 endpoint. The
token generation occurs inside the Sapphire's TEE and the communication
with your Web3 endpoint is secured via HTTPS.
```
